src/features/tracks/hooks/use-search-params-state.tsx

```diff
@@ -0,0 +1,56 @@
+import { getRouteApi } from '@tanstack/react-router';
+import { PaginationState, SortingState } from '@tanstack/react-table';
+import { isValidSortValue } from '../lib/type-guards';
+
+export const useSearchParamsState = () => {
```

Comment:
Could be more specific name as you handle specific params in your case such as pagination, query, search